roles/openafs_server/tasks/server-key.yaml

```diff
@@ -23,17 +21,16 @@
     dest: "{{ afs_afsconfdir }}/rxkad.keytab"
     remote_src: yes
   register: keytab
-  notify:
-    - Restart OpenAFS servers
-
+  
 - name: Check for akeyconvert availability
   stat:
-    path: "{{ afs_akeyconvert }}"
-  register: akeyconvert
+    path: "{{ afs_asetkey }}"
+  register: asetkey
   changed_when: False
 
 - name: Set the service key
   become: yes
-  command: "{{ afs_akeyconvert }}"
-  register: akeyconvert_results
-  when: keytab.changed and akeyconvert.stat.exists
+  command: "{{ afs_asetkey }} add 1 {{ afs_afsconfdir }}/rxkad.keytab afs/{{ afs_cell }}" 
```

Comment:
This only works for DES keys, because you need to specify the key enctype etc for other keys.

For DES keys we would not make a rxkad.keytab i think. The rxkad.keytab was for non-DES keys in 1.6.5 to up to 1.8.0.